_pytest/recwarn.py

```diff
@@ -110,15 +110,10 @@ def warns(expected_warning, *args, **kwargs):
             return func(*args[1:], **kwargs)
 
 
-class RecordedWarning(object):
-    def __init__(self, message, category, filename, lineno, file, line):
-        self.message = message
-        self.category = category
-        self.filename = filename
-        self.lineno = lineno
-        self.file = file
-        self.line = line
-
+RecordedWarning = namedtuple('RecordedWarning', (
```

Comment:
There's a small side effect here, `RecordedWarning` attributes are now read-only.

The docs themselves have this to say about it:

```
Each recorded warning has the attributes message, category, filename, lineno, file, and line. The category is the class of the warning. The message is the warning itself; calling str(message) will return the actual message of the warning.
```

IMO your change is good and I doubt there are people writing to the attributes, but I thought I would mention this anyway.